src/Flagged/Modal.jsx

```diff
@@ -0,0 +1,71 @@
+let {
+  onReport,
+  reportedAccountId,
+  contentModerationFlagValue,
+  ...forwardedProps
+} = props;
+
+const profileUrl = `#/${REPL_ACCOUNT}/widget/ProfilePage?accountId=${reportedAccountId}`;
+
+let getFlaggedAccountsList;
+
+// copied from Moderate.jsx so maybe it's wrong
+if (forwardedProps.open) {
+  getFlaggedAccountsList = Social.get(
+    `${context.accountId}/moderate/users`,
+    "optimistic"
+  );
+}
+
+const socialSet = (index) =>
+  Social.set(index, {
+    onCommit: () => {
+      onReport && onReport();
+    },
+    onCancel: () => onOpenChange(false),
+  });
+
+const reportAccount = () => {
+  socialSet({
+    index: {
+      flag: JSON.stringify({
+        key: "main",
+        value: contentModerationFlagValue,
+      }),
+    },
+  });
+};
+
+const reportAccountWithPosts = () => {
+  const flaggedAccountsList = getFlaggedAccountsList
+    ? JSON.parse(getFlaggedAccountsList)
+    : [];
+  const flaggedAccountsListSet = new Set(flaggedAccountsList);
+  const filterFlaggedAccounts = [
+    ...flaggedAccountsListSet.add(reportedAccountId),
+  ];
+
+  socialSet({
+    index: {
+      moderate: {
+        [filterFlaggedAccounts]: "",
```

Comment:
Let's make this `report` for now. It can get more options when we do the reporting flow. In that flow `report` will be the value to store for a generic report (without choosing a reason).
```suggestion
        [filterFlaggedAccounts]: "report",
```